[Lee Jun Heng] iP

Junixm opened the pull request:
# 📋 **Treky**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Treky is a chatbot that helps you manage and tracks your task easily.

- text-based
- easy to learn
- SUPER FAST to use

## **Features**
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

### **Releases**
1. Download it [here](https://github.com/Junixm/ip/releases/tag/A-Jar)
2. Run using this command:
```java
java -jar shin.jar
``` 
3. Add tasks, deadlines and events using simple commands.
4. Manage your tasks effortlessly.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Junixm committed:
Renamed main class

Junixm committed:
Added greeting and farewell

Junixm committed:
Added support for echo

Junixm committed:
Added Task class

Junixm committed:
Added support for addTask

Junixm committed:
Added support for listTask

Junixm committed:
Add support to mark and unmark in Task class

Junixm committed:
Add TaskManager class

Junixm committed:
Moved task methods to TaskManager class

Junixm committed:
add support to mark and unmark task

Junixm committed:
add support for todo, event and deadline task

Junixm committed:
Fixed sc no line bug

Junixm committed:
add automate UI Testing

Junixm committed:
add delete feature

Junixm committed:
refactor Task.java

Junixm committed:
Refactor Treky.java

Junixm committed:
Add TrekyException.java

Junixm committed:
Refactor Task.java

Junixm committed:
Refactor TaskManager.java

Junixm committed:
Add CommandHandler.java

Junixm committed:
Update test files

Junixm committed:
Add feature toFileString

Junixm committed:
Refactor Treky.java: Add Storage class

Junixm committed:
Add Storage class

Junixm committed:
Add getTasks

Junixm committed:
Add support for Storage class

Junixm committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add support for Storage class
  Add getTasks
  Add Storage class
  Refactor Treky.java: Add Storage class
  Add feature toFileString

Junixm committed:
Add ui class

Junixm committed:
Refactor Treky class

Junixm committed:
Refactor CommandHandler

Junixm committed:
Refactor Storage class

Junixm committed:
Refactor Treky class

Junixm committed:
Refactor execute method

Junixm committed:
Refactor code

Junixm committed:
Refactor code

Junixm committed:
Refactor code to use LocalDate

Junixm committed:
Refactor code

Junixm committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Refactor code
  Refactor code to use LocalDate
  Refactor code
  Refactor code
  Refactor execute method
  Refactor Treky class

Junixm committed:
Move code to treky.task package

Junixm committed:
Move code to treky package

Junixm committed:
Move code to treky.command package

Junixm committed:
Refactor test

Junixm committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Junixm committed:
Add support getDone

Junixm committed:
Change method name toFileString to toSaveString

Junixm committed:
Refactor output to ui

Junixm committed:
Refactor output to ui

Junixm committed:
Modify return List instead of ArrayList

Junixm committed:
Add support showResult

Junixm committed:
Refactor code

Junixm committed:
Tweak gradle build

Junixm committed:
Add JUnit test for CommandHandler

Junixm committed:
Add JUnit test for TaskManager

samuelneo commented on src/main/java/treky/command/CommandHandler.java:
```diff
@@ -0,0 +1,164 @@
+
+    private String addResult(Task task) {
+        return "Got it. I've added this task:\n  " + task + "\n Now you have " +
+                taskManager.getTaskListSize() + " tasks in the list.";
```
coding standard: line break before an operator (in this case, the `+`)
I noticed the same issue in several other places too

samuelneo commented on src/main/java/treky/command/CommandHandler.java:
```diff
@@ -0,0 +1,164 @@
+            return addResult(task);
+        } catch (DateTimeParseException e) {
+            throw new TrekyException("Invalid date format. Please use yyyy-MM-dd.");
+        }
```
perhaps the date/time parsing functionality can be extracted to another class? this would also make it easier to implement more date/time features (e.g. supporting time or date+time, allowing more formats)

samuelneo commented on src/main/java/treky/Treky.java:
```diff
@@ -0,0 +1,49 @@
+import treky.command.Storage;
+import treky.command.TaskManager;
+
+public class Treky {
```
coding standard: javadoc is required for public classes and public methods (though I've noticed you haven't committed `A-JavaDoc` yet, so I presume this would be fixed with that commit)

samuelneo commented on src/main/java/treky/TrekyException.java:
```diff
@@ -0,0 +1,7 @@
+package treky;
+
+public class TrekyException extends Exception {
```
seems like a catch-all exception class, maybe having subclasses of this (e.g. `TrekyInvalidCommandException`, etc.) may be useful?

samuelneo reviewed (commented):
nice, clean code

Junixm committed:
Add JavaDoc

Junixm committed:
Add JavaDoc

Junixm committed:
Add JavaDoc

Junixm committed:
Add JavaDoc

Junixm committed:
Update tests

Junixm committed:
Add JavaDoc

Junixm committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add JavaDoc
  Update tests
  Add JavaDoc
  Add JavaDoc
  Add JavaDoc
  Add JavaDoc

Junixm committed:
Merge branch 'master' into branch-A-CodingStandard

* master:
  Add JavaDoc
  Update tests
  Add JavaDoc
  Add JavaDoc
  Add JavaDoc
  Add JavaDoc

Junixm committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:

Junixm committed:
Merge branch 'master' into branch-Level-9

* master:
  Add JavaDoc
  Update tests
  Add JavaDoc
  Add JavaDoc
  Add JavaDoc
  Add JavaDoc

Junixm committed:
Add support for getting description

Junixm committed:
Add support for find task using keywords

Junixm committed:
Add support for find feature

Junixm committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add support for find feature
  Add support for find task using keywords
  Add support for getting description

Junixm committed:
Fix find task bug

Junixm committed:
Add JavaDoc for class

chenxu20 commented on src/main/java/treky/command/CommandHandler.java:
```diff
@@ -0,0 +1,201 @@
+    }
+
+    private String execute(String command, String description) throws TrekyException {
+        return switch (command.toLowerCase()) {
+        case "bye" -> setExit();
+        case "todo" -> addTodo(description);
+        case "deadline" -> addDeadline(description);
+        case "event" -> addEvent(description);
+        case "list" -> listTasks();
+        case "mark" -> markTask(description, true);
+        case "unmark" -> markTask(description, false);
+        case "delete" -> deleteTask(description);
+        case "find" -> findTask(description);
+        default -> throw new TrekyException("I'm sorry, but I don't know what that means :-(");
```
Should the 'cases' and 'default' lines be indented?

chenxu20 commented on src/main/java/treky/command/CommandHandler.java:
```diff
@@ -0,0 +1,201 @@
+    }
+
+    private String addDeadline(String description) throws TrekyException {
+        if (description.isEmpty()) {
+            throw new TrekyException("Format: deadline <description> /by <yyyy-MM-dd>");
+        }
+        String[] parts = description.split("/by");
+        String trimmedDescription = parts[0].trim();
+        if (trimmedDescription.isEmpty()) {
+            throw new TrekyException("Description cannot be empty.\n" +
+                    "Format: deadline <description> /by <yyyy-MM-dd>");
+        }
+        if (parts.length != 2 || parts[1].trim().isEmpty()) {
+            throw new TrekyException("Deadline cannot be empty\n" +
+                    "Format: deadline <description> /by <yyyy-MM-dd>");
+        }
```
Perhaps adding some empty lines between blocks of code can help with overall cleanliness. Noticed this in other parts of your code as well.

chenxu20 commented on src/main/java/treky/command/Storage.java:
```diff
@@ -0,0 +1,143 @@
+        String type = parts[0];
+        boolean isDone = parts[1].equals("1");
+        String description = parts[2];
+        return switch (type) {
+            case "T" -> {
+                if (parts.length != 3) {
```
Nice indentation!

chenxu20 commented on src/main/java/treky/command/TaskManager.java:
```diff
@@ -0,0 +1,128 @@
+import java.util.ArrayList;
+import java.util.List;
+
+import treky.task.Task;
+import treky.task.Todo;
+import treky.task.Deadline;
+import treky.task.Event;
```
Perhaps you can arrange the import statements in alphabetical order.

chenxu20 commented on src/main/java/treky/task/Deadline.java:
```diff
@@ -0,0 +1,43 @@
+     * @param date        Date of task.
+     * @param isDone      Status of task.
+     */
+    public Deadline(String description, LocalDate date, boolean isDone) {
+        super(description, isDone);
+        this.date = date;
```
I like how you have 2 constructors here.

chenxu20 reviewed (commented):
LGTM.

Junixm committed:
Move class to exception package

Junixm committed:
Add fatal exception class

Junixm committed:
Move class to storage package

Junixm committed:
Move class to ui package

Junixm committed:
Rename packages

Junixm committed:
Rename TaskManager to TaskList

Junixm committed:
Remove test

Junixm committed:
Refactor code

Junixm committed:
Add task list encoder and decoder

Junixm committed:
Remove test

Junixm committed:
Tweak comments

Junixm committed:
Add executable interface

Junixm committed:
Update ignore file

Junixm committed:
Refactor code for easier readability

Junixm committed:
Add featured command classes

Junixm committed:
Add featured command classes

Junixm committed:
Add executable interface

Junixm committed:
Tweak exit method

Junixm committed:
Fix unable to write to file bug

Junixm committed:
Add test

Junixm committed:
Refactor code: split command functions into classes

Junixm committed:
Refactor code to improve readability

Junixm committed:
Change class name to TextUi

Junixm committed:
Tweak layout

Junixm committed:
Add images

Junixm committed:
Add css files

Junixm committed:
Add fxml files

Junixm committed:
Add gui function

Junixm committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add gui function
  Add fxml files
  Add css files
  Add images
  Tweak layout
  Change class name to TextUi

Junixm committed:
Add assertions

Added assertions in all command, task and TaskList classes to validate
assumptions and catch potential bugs.

Junixm committed:
Merge pull request #1 from Junixm/branch-A-Assertions

Add assertions

Junixm committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Add assertions

Junixm committed:
Refactor code

Junixm committed:
Merge pull request #2 from Junixm/branch-A-CodeQuality

Branch a code quality

Junixm committed:
Add alias feature to support user-defined command shorts

Users currently have to type long commands repeatedly, which can be
inefficient and cumbersome.

Introduce an alias feature to allow users to define custom shortcuts for
commands.

Implement methods for creating aliases. Modify the commandHandler parse
method to revert aliases to original command before preparing command.

This approach ensures users can easily access frequently used commands
with shorter aliases, improving overall user experience.

Aliases is not saved when closed.
Deleting aliases is not yet supported.

Junixm committed:
Fix bug: Correct chat bubble orientation

Junixm committed:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  Fix bug: Correct chat bubble orientation
  Add alias feature to support user-defined command shorts

Junixm committed:
Tweak gui

Junixm committed:
Merge branch 'branch-A-BetterGui'

* branch-A-BetterGui:
  Tweak gui

Junixm committed:
Add pic

Junixm committed:
Update README.md

Junixm committed:
Refactor alias class name

Junixm committed:
Add user guide

Junixm committed:
Restore readme

Junixm committed:
Update README.md

Junixm committed:
Update README.md

Junixm committed:
Update README.md

Junixm committed:
Update README.md

Junixm committed:
Update README.md

Junixm committed:
Update README.md

Junixm committed:
Update README.md

Junixm committed:
Update README.md
